src/hazmat/float.rs

```diff
@@ -160,4 +219,90 @@ mod tests {
             assert_eq!(test, reference);
         }
     }
+
+    #[test]
+    fn test_ln_x_small() {
+        let test_cases: Vec<(u128, f64)> = vec![
+            (1, 0.0),
+            (3, 1.09861228866811),
+            (4, 1.3862943611198906),
+            (5, 1.6094379124341004),
+            (10u128.pow(2), 4.60517018598809),
+            (10u128.pow(3), 6.90775527898214),
+            (10u128.pow(4), 9.21034037197618),
+            (10u128.pow(5), 11.51292546497023),
+            (10u128.pow(6), 13.81551055796427),
+            (10u128.pow(7), 16.11809565095832),
+            (10u128.pow(8), 18.42068074395237),
+            (10u128.pow(9), 20.72326583694641),
+            (10u128.pow(10), 23.02585092994046),
+            (10u128.pow(11), 25.3284360229345),
+            (10u128.pow(12), 27.63102111592855),
+            (10u128.pow(13), 29.93360620892259),
+            (10u128.pow(14), 32.23619130191664),
+            (10u128.pow(15), 34.53877639491069),
+            (10u128.pow(16), 36.84136148790473),
+            (10u128.pow(17), 39.14394658089878),
+            (10u128.pow(18), 41.44653167389282),
+            (10u128.pow(19), 43.74911676688687),
+            (10u128.pow(20), 46.05170185988091),
+            (10u128.pow(21), 48.35428695287496),
+            (10u128.pow(22), 50.65687204586901),
+            (10u128.pow(23), 52.95945713886305),
+            (10u128.pow(24), 55.262042231857096),
+            (10u128.pow(25), 57.56462732485114),
+            (10u128.pow(26), 59.86721241784519),
+            (10u128.pow(27), 62.16979751083923),
+            (10u128.pow(28), 64.47238260383328),
+            (10u128.pow(29), 66.77496769682733),
+            (10u128.pow(30), 69.07755278982137),
+            (10u128.pow(31), 71.38013788281542),
+            (10u128.pow(32), 73.68272297580946),
+            (10u128.pow(33), 75.9853080688035),
+            (10u128.pow(34), 78.28789316179756),
+            (10u128.pow(35), 80.5904782547916),
+            (10u128.pow(36), 82.89306334778564),
+            (10u128.pow(37), 85.19564844077969),
+            (10u128.pow(38), 87.49823353377374),
+        ];
+        for (x, expected) in test_cases.iter() {
+            let x_big = U128::from_u128(*x);
+            let result = ln(&x_big);
+            assert!(
+                (result - *expected).abs() < f64::EPSILON * 100.0,
+                "x: {x}, mine: {result}, expected: {expected}"
+            );
+        }
+    }
+
+    #[test]
+    fn test_ln_x_large() {
+        let mut rng = rand_chacha::ChaCha8Rng::from_seed(*b"01234567890123456789012345678901");
+        let expected_256 = 172.29242258375361;
+
+        let x = U256::random(&mut rng);
```

Comment:
Why note just hardcode `x` instead of hardcoding the random seed?

Reply:
Just because it's shorter and saves me `eprintln!`-ing the output, run the test and then paste in the value. But sure, might be more elegant and readable.